File: Injections.NullModules/NullGPIO.cs
using System;
using TinyIoC;
using ACRemote;

namespace Injections.NullModules {
	public class NullGPIO : Defaultable, IPlatformGPIO {
		System.Collections.Generic.Dictionary<int, bool> pins = new System.Collections.Generic.Dictionary<int, bool>();
		public static void register(TinyIoCContainer kern) {
			if (!non_default_exists<IPlatformGPIO>(typeof(NullGPIO))) {
				kern.Register<IPlatformGPIO, NullGPIO>().AsSingleton();
			}
		}
		public void setup(int pin, direction dir) {
			pins[pin] = false;
		}
		public bool this[int val] {
			get {
				return pins[val];
			}
			set {
				pins[val] = value;
			}
		}
	}
}

File: Injections.NullModules/NullGPIO.vb
' Null GPIO module uses a map to store state
' Author: Kevin Ross

Imports TinyIoC
Imports ACRemote

Public Class NullGPIO
    Inherits Defaultable
    Implements IPlatformGPIO
    Private pins As New System.Collections.Generic.Dictionary(Of Integer, Boolean)()
    Public Sub setup(pin As Integer, dir As direction) Implements IPlatformGPIO.setup
        pins(pin) = False
        Console.WriteLine(String.Format("GPIO: setup pin {0} as {1}", pin, dir.ToString()))
    End Sub
    Default Public Property Item(val As Integer) As Boolean Implements IPlatformGPIO.Item
        Get
            Console.WriteLine(String.Format("GPIO: get pin {0}={1}", val, pins(val)))
            Return pins(val)
        End Get
        Set
            Console.WriteLine(String.Format("GPIO: set pin {0}={1}", val, Value))
            pins(val) = Value
        End Set
    End Property
End Class